src/compiler/performance.ts

```diff
@@ -53,9 +50,8 @@ namespace ts.performance {
      * @param markName The name of the mark.
      */
     export function mark(markName: string) {
-        if (enabled) {
-            marks.set(markName, timestamp());
-            counts.set(markName, (counts.get(markName) || 0) + 1);
+        if (perfHooks && enabled) {

```

Comment:
Yes. 
```ts
ts.performance.enable(); // creates `perfHooks`, sets `enabled` to `true`, calls `perfObserver.connect()`
ts.performance.disable(); // sets `enabled` to `false`, calls `perfObserver.disconnect()`
ts.performance.enable(); // `perfHooks` already exists, sets `enabled` to `true`, calls `perfObserver.connect()`
```
No need to recreate the shim every time we disable/enable (should we ever chose to do so)

Reply:
If `enabled` implies `!!perfHooks`, then it seems like we shouldn't need to check `perfHooks`.

Reply:
I check for `perfHooks` because our type system doesn't *know* that `enabled` implies `!!perfHooks`. The alternative is to write `perfHooks!.` everywhere, which could be hazardous if we make any subtle code changes later. I'm generally not a fan of unchecked casts.